kexdh.c

```diff
@@ -83,24 +90,92 @@ kex_dh_compute_key(struct kex *kex, BIGNUM *dh_pub, struct sshbuf *out)
 		r = SSH_ERR_MESSAGE_INCOMPLETE;
 		goto out;
 	}
-	klen = DH_size(kex->dh);
+
+	DH_get0_pqg(kex->dh, &p, &q, &g);
```

Comment:
I would  move this just before the BN_dup()

Reply:
done. Also added some checks